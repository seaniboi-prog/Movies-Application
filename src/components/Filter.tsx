import React from 'react'
import './mainPage.scss';

interface Props {
    filterQuery: string;
    setFilterQ: React.Dispatch<React.SetStateAction<string>>;
    genreTypes: string[];
}

const Filter:React.FC<Props> = ({filterQuery, setFilterQ, genreTypes}: Props) => {

    return (
    <div className='filter__bar'>
        <select value={filterQuery} onChange={e => setFilterQ(e.target.value)}>
            <option value="all">All</option>
        { genreTypes.map(g => <option value={g}>{g[0].toUpperCase() + g.slice(1)}</option>) } 
        </select>
    </div>
    );
}

export default Filter;