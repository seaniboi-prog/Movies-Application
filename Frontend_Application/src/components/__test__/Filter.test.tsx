import Filter from '../Filter'

import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer';

describe('Filter Component Tests', () => {

    const genreValue:string = "All";
    const dummyGenres:string[] = ["Genre1", "Genre2", "Genre3"];

    const renderComponent = () => {
        render(<Filter filterQuery={genreValue} setFilterQ={()=>{}} genreTypes={dummyGenres} />);
    }

    beforeEach(renderComponent);

    afterEach(cleanup);

    it("Renders Filter Component correctly", () => {
        const filterComponent:HTMLElement = screen.getByTestId('filter');
        expect(filterComponent).toBeInTheDocument();
        expect(filterComponent).toHaveTextContent("Genre");
    });

    it("Renders select element correclty", () => {
        const selectElement:HTMLElement = screen.getByTestId('filter-select');
        expect(selectElement).toBeInTheDocument();
    });

    it("Renders option elements correclty", () => {
        const allElement:HTMLElement = screen.getByTestId('all-option');
        expect(allElement).toBeInTheDocument();
        expect(allElement).toHaveTextContent("All");
        expect(allElement.getAttribute('value')).toBe('all');

        const optionElements:HTMLElement[] = screen.getAllByTestId('list-option');
        expect(optionElements).toHaveLength(dummyGenres.length);
    });

    it("Matches Snapshot", () => {
        const tree = renderer.create(<Filter filterQuery={genreValue} setFilterQ={()=>{}} genreTypes={dummyGenres}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
});