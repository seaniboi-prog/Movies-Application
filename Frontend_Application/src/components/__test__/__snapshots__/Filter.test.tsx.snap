// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component Tests Matches Snapshot 1`] = `
<div
  className="filter__bar"
  data-testid="filter"
>
  Genre
  <select
    data-testid="filter-select"
    onChange={[Function]}
    value="All"
  >
    <option
      data-testid="all-option"
      value="all"
    >
      All
    </option>
    <option
      data-testid="list-option"
      value="Genre1"
    >
      Genre1
    </option>
    <option
      data-testid="list-option"
      value="Genre2"
    >
      Genre2
    </option>
    <option
      data-testid="list-option"
      value="Genre3"
    >
      Genre3
    </option>
  </select>
</div>
`;
